specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json

```diff
@@ -551,31 +550,395 @@
+                {
+                  "name": "RecoverDeletedHsmKey",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/deletedKeys/recover/action",
+                  "description": "Recover deleted HSM key."
+                },
+                {
+                  "name": "BackupHsmKeys",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/backup/action",
+                  "description": "Backup HSM keys."
+                },
+                {
+                  "name": "RestoreHsmKey",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/restore/action",
+                  "description": "Restore HSM keys."
+                },
+                {
+                  "name": "DeleteRoleAssignment",
+                  "value": "Microsoft.KeyVault/managedHsm/roleAssignments/delete/action",
+                  "description": "Delete role assignment."
+                },
+                {
+                  "name": "GetRoleRoleAssignment",
+                  "value": "Microsoft.KeyVault/managedHsm/roleAssignments/read/action",
+                  "description": "Get role assignment."
+                },
+                {
+                  "name": "WriteRoleAssignment",
+                  "value": "Microsoft.KeyVault/managedHsm/roleAssignments/write/action",
+                  "description": "Create or update role assignment."
+                },
+                {
+                  "name": "ReadRoleDefinition",
+                  "value": "Microsoft.KeyVault/managedHsm/roleDefinitions/read/action",
+                  "description": "Get role definition."
+                },
+                {
+                  "name": "EncryptHsmKeys",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/encrypt/action",
+                  "description": "Encrypt HSM keys."
+                },
+                {
+                  "name": "DecryptHsmKeys",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/decrypt/action",
+                  "description": "Decrypt HSM keys."
+                },
+                {
+                  "name": "WrapHsmKeys",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/wrap/action",
+                  "description": "Wrap HSM keys."
+                },
+                {
+                  "name": "UnwrapHsmKeys",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/unwrap/action",
+                  "description": "Unwrap HSM keys."
+                },
+                {
+                  "name": "SignHsmKeys",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/sign/action",
+                  "description": "Sign HSM keys."
+                },
+                {
+                  "name": "VerifyHsmKeys",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/verify/action",
+                  "description": "Verify HSM keys."
+                },
+                {
+                  "name": "CreateHsmKey",
```

Comment:
Since we also have "write", does this truly mean a subset of permissions, i.e. can *only create, not update*? Just wanted to make sure. Less about the swagger change and more about the service/granularity.

Reply:
"write" only grants access to writing / updating key metadata (e.g. tags) whereas create allows for creation of new keys